packages/flutter/lib/src/widgets/safe_area.dart

```diff
@@ -25,6 +25,23 @@ import 'media_query.dart';
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=lkF0TQJO0bA}
```

Comment:
This "widget of the week" video gives a really nice overview, but in my mind there are some unanswered questions:
- Is it good to give a safe area for both a scaffold's `appBar` and its `body`?
- What about when a screen is built without using a `Scaffold`?
(e.g. if a widget only covers the bottom half of the screen, do we need to set `top: false` or will Flutter automatically take care of it?)

Reply:
@QuncCccccc do you know the answer to these questions & what the best practices are for Material and `SafeArea`?

Reply:
> Is it good to give a safe area for both a scaffold's appBar and its body?

If we have a Scaffold and have body only (without an app bar or bottom navigation bar), it's good to have a safe area to avoid intrusions, see [doc](https://github.com/flutter/flutter/blob/21814805c1ba74f36a37f73203f34179c24a8584/packages/flutter/lib/src/material/scaffold.dart#L1579) here. If we have Scaffold.appBar, [the safe area is taken care of by it](https://github.com/flutter/flutter/blob/21814805c1ba74f36a37f73203f34179c24a8584/packages/flutter/lib/src/material/app_bar.dart#L114), so we don't need to worry about the top of body, but if we want the bottom/left/right of the body to be inside of safe area, we should use `SafeArea`.

> What about when a screen is built without using a Scaffold?
(e.g. if a widget only covers the bottom half of the screen, do we need to set top: false or will Flutter automatically take care of it?)

Sorry I'm mot super clear about the answer here. But for the example question, yes, we want to set `top: false` even if the widget only covers half the screen. 

Please let me know if there's any questions or anything doesn't make sense:)!

Reply:
Thanks very much for answering my questions!

> If we have Scaffold.appBar, [the safe area is taken care of by it](https://github.com/flutter/flutter/blob/21814805c1ba74f36a37f73203f34179c24a8584/packages/flutter/lib/src/material/app_bar.dart#L114), so we don't need to worry about the top of body

That's great to hear! I wonder if there's a good spot for this info in the documentation.

Reply:
> Just feel `SafeArea` is typically useful on mobile devices, like the "notch".

100% agree; setting a `SafeArea` is pretty useless on web.
However: I think we can still demo this by making our own "notch" with custom `MediaQueryData` :)

Reply:
> I went ahead and made a [DartPad sample](https://dartpad.dev/?id=29cb777b13b9656b05c8230fb9a0a421&channel=main) since I thought it'd be fun. Let me know what you think!

I think the example is very interesting and great to have as a DartPad sample🤩! I'm thinking whether it makes sense to have both since the code snippet is very straightforward and easy to understand.

Reply:
> the code snippet is very straightforward and easy to understand.

True that, I put waay too much effort into "optimizing rebuilds" just for a demo file 😅

<br>

I've found that ~20 seconds of messing around with an interactive sample is much better at building the right intuition than ~2 minutes of reading through paragraphs of documentation.

I definitely agree that including both the snippet and the DartPad sample is something worth considering, since the snippet is indeed very simple & straightforward 👍